nomad/structs/csi.go

```diff
@@ -846,9 +846,13 @@ func (v *CSIVolume) Merge(other *CSIVolume) error {
 			"volume parameters cannot be updated"))
 	}
 
-	// Context is mutable and will be used during controller
-	// validation
-	v.Context = other.Context
+	// Context is mutable and will be used during controller validation, but we
+	// need to ensure we don't remove context that's been previously stored
+	// server-side if the user has submitted an update without adding it to the
+	// spec manually (which we should not require)
+	if len(other.Context) != 0 {
+		v.Context = other.Context
+	}
```

Comment:
this makes good sense for `create`, where the controller plugin generates the `context` dynamically, but I wonder about this edge case:

what if a user `register`s a volume with some `context`, but later wants to remove the context entirely? is that even a plausible concern? I suppose one could `deregister` the volume then re-`register` it with no context, since the actual volume out in the world would be unaffected.

Reply:
The only time that would happen is if they updated the plugin and the new version of the plugin _requires_ no context. In that case, you're almost certainly better off deregistering and re-registering